optimizely/decision/composite_service.go

```diff
@@ -31,15 +31,15 @@ var csLogger = logging.GetLogger("CompositeDecisionService")
 type CompositeService struct {
 	// experimentDecisionServices []ExperimentDecisionService
 	featureDecisionServices []FeatureService
-	notificationCenter      notification.Center
+	sdkKey                  string
 }
 
 // NewCompositeService returns a new instance of the DefeaultDecisionEngine
-func NewCompositeService(notificationCenter notification.Center) *CompositeService {
+func NewCompositeService(sdkKey string) *CompositeService {
```

Comment:
The one caveat about this impl is that the component needs to know about the SDK Key, which I don't think it might be too much to ask since customers are already providing it to the client so even if they instantiate a new composite service outside the client, they'll still have access to the SDK Key